src/Passes/Source/RuleTransformationPass/RulePass.hpp

```diff
@@ -0,0 +1,245 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Logging/ILogger.hpp"
+#include "Profile/Profile.hpp"
+#include "RuleTransformationPass/RuleTransformationPassConfiguration.hpp"
+#include "Rules/RuleSet.hpp"
+#include "Types/Types.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <functional>
+#include <unordered_map>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    /// This class applies a set of transformation rules to the IR to transform it into a new IR. The
+    /// rules are added using the RuleSet class which allows the developer to create one or more rules
+    /// on how to transform the IR.
```

Comment:
> This class applies a set of transformation rules

This comment tells me that 
the class should be named something like `TransformationRulesPass` (a pass of the transformation rules) 
rather than `RuleTransformationPass` (a pass that transforms the rules).

Reply:
Renamed.